src/group_theory/order_of_element.lean

```diff
@@ -103,40 +103,189 @@ lemma card_quotient_dvd_card (s : subgroup α) [decidable_pred (λ a, a ∈ s)]
   fintype.card (quotient s) ∣ fintype.card α :=
 by simp [card_eq_card_quotient_mul_card_subgroup s]
 
-lemma exists_gpow_eq_one (a : α) : ∃i≠0, a ^ (i:ℤ) = 1 :=
-have ¬ injective (λi:ℤ, a ^ i),
-  from not_injective_infinite_fintype _,
-let ⟨i, j, a_eq, ne⟩ := show ∃(i j : ℤ), a ^ i = a ^ j ∧ i ≠ j,
-  by rw [injective] at this; simpa [not_forall] in
-have a ^ (i - j) = 1,
-  by simp [sub_eq_add_neg, gpow_add, gpow_neg, a_eq],
-⟨i - j, sub_ne_zero.mpr ne, this⟩
-
-lemma exists_pow_eq_one (a : α) : ∃i > 0, a ^ i = 1 :=
-let ⟨i, hi, eq⟩ := exists_gpow_eq_one a in
+end order_of
+
+section order_of
+
+section monoid
+open_locale classical
+variables {α} [monoid α] [decidable_eq α]
+
+
+/-- `order_of a` is the order of the element `a`, i.e. the `n ≥ 1`, s.t. `a ^ n = 1` if it exists.
+Otherwise, i.e. if `a` is of infinite order, then `order_of a` is `0` by convention.-/
+def order_of (a : α) [decidable_pred (λ n, 0 < n ∧ a ^ n = 1)] [decidable (∃ n, 0 < n ∧ a ^ n = 1)] : ℕ :=
```

Comment:
I realize now that these long decidable arguments are mostly pointless unless you repeat them on the lemmas below and also remove the `open_locale classical`.

So perhaps put these up in `variables`?

Reply:
@eric-wieser Any idea what makes the proof not work anymore that made this run fail?